examples/server/server.cpp

```diff
@@ -603,14 +608,451 @@ void get_req_parameters(const Request & req, whisper_params & params)
+                for (int i = 0; i < n_segments; ++i) {
+                    json segment = json{
+                        {"id", i},
+                        {"text", whisper_full_get_segment_text(ctx_, i)},
+                    };
+
+                    if (!task.params.no_timestamps) {
+                        segment["start"] = whisper_full_get_segment_t0(ctx_, i) * 0.01;
+                        segment["end"] = whisper_full_get_segment_t1(ctx_, i) * 0.01;
+                    }
+
+                    float total_logprob = 0;
+                    const int n_tokens = whisper_full_n_tokens(ctx_, i);
+                    for (int j = 0; j < n_tokens; ++j) {
+                        whisper_token_data token = whisper_full_get_token_data(ctx_, i, j);
+                        if (token.id >= whisper_token_eot(ctx_)) {
+                            continue;
+                        }
+
+                        segment["tokens"].push_back(token.id);
+                        json word = json{{"word", whisper_full_get_token_text(ctx_, i, j)}};
+                        if (!task.params.no_timestamps) {
+                            word["start"] = token.t0 * 0.01;
+                            word["end"] = token.t1 * 0.01;
+                            word["t_dtw"] = token.t_dtw;
+                        }
+                        word["probability"] = token.p;
+                        total_logprob += token.plog;
+                        segment["words"].push_back(word);
+                    }
+
+                    segment["temperature"] = task.params.temperature;
+                    segment["avg_logprob"] = total_logprob / n_tokens;
+                    segment["no_speech_prob"] = whisper_full_get_segment_no_speech_prob(ctx_, i);
+
+                    jres["segments"].push_back(segment);
+                }
+                result.output = jres.dump(-1, ' ', false, json::error_handler_t::replace);
+                result.content_type = "application/json";
+            }
+            else {
+                std::string results = output_str(ctx_, task.params, task.pcmf32s);
+                json jres = json{{"text", results}};
+                result.output = jres.dump(-1, ' ', false, json::error_handler_t::replace);
+                result.content_type = "application/json";
+            }
+
+            result.success = true;
+            result.status_code = 200;
+        }
+        catch (const std::exception& e) {
+            result.success = false;
+            result.error_msg = "{\"error\":\"" + std::string(e.what()) + "\"}";
+            result.status_code = 500;
+        }
+
+        return result;
+    }
+
+    struct whisper_context* ctx_;
+    std::vector<std::thread> workers_;
+    std::queue<std::shared_ptr<WhisperTask>> tasks_;
+    std::mutex queue_mutex_;
+    std::condition_variable queue_cv_;
+    std::mutex whisper_mutex_; // Protect whisper context access
+    std::atomic<bool> stop_flag_;
```

Comment:
I think `stop_flag_` could just be a bool as the accesses to it are all guarded by the `queue_cv_` lock.